components/avatar/avatar.vue

```diff
@@ -203,20 +204,26 @@ export default {
     },
   },
 
-  mounted () {
-    this.init();
+  watch: {
+    defaultSlotContent: {
+      deep: true,
+      immediate: true,
+      handler: function () {
+        this.init();
+      },
+    },
   },
 
   updated () {
-    if (this.kind === 'initials') {
-      this.slottedInitials = this.$slots.default[0].text || this.$slots.default[0].textContent;
-      this.formatInitials(this.slottedInitials);
-    }
+    this.defaultSlotContent = this.$slots.default ? this.$slots.default[0] : null;
+    this.imageLoadedSuccessfully = this.defaultSlotContent?.elm?.complete;
   },
 
   methods: {
     init () {
-      const firstChild = this.$refs.canvas.firstChild;
+      this.defaultSlotContent = this.$slots.default ? this.$slots.default[0] : null;
+      const firstChild = this.defaultSlotContent?.elm ? this.defaultSlotContent.elm : this.defaultSlotContent;
+      this.formatInitials(this.initials);
```

Comment:
This wasn't working correctly because the $ref.canvas content was changing with the 'kind' property, sometimes overriding the checks I was doing later so the component didn't detected the changes properly.

This way, we're checking the actual default slot content property to decide what we should render.